### PR TITLE
Implement draw structs

### PR DESCRIPTION
**Pull request by duncanspumpkin**
This is the main conversion of the paint struct graph into draw calls.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Could do with a bit of testing.

File: src/OpenLoco/src/Paint/Paint.cpp [hunk 1]
```diff
@@ -761,10 +761,202 @@ namespace OpenLoco::Paint
...
+                {
+                    if (cullPaintStructImage(attachPs->imageId, _viewFlags))
+                    {
+                        continue;
+                    }
+                }
+                drawAttachStruct(rt, drawingCtx, *ps, *attachPs, shouldCullAttach);
+            }
+        }
+        // call(0x0045EA23);
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Paint/Paint.cpp:959]: remove

**Review by AaronVanGeffen — APPROVED** [review]
Rendering looks alright, as far as I can tell. There seems to be a minor impact on performance (FPS) on my old system, but it's hard to measure. Nothing significant, anyway.

**Comment by duncanspumpkin** [comment]
> Rendering looks alright, as far as I can tell. There seems to be a minor impact on performance (FPS) on my old system, but it's hard to measure. Nothing significant, anyway.

Really a performance impact? @ZehMatt got a performance gain.

File: src/OpenLoco/src/Paint/Paint.cpp [hunk 2]
```diff
@@ -761,10 +761,201 @@ namespace OpenLoco::Paint
...
+
     // 0x0045EA23
     void PaintSession::drawStructs()
     {
-        call(0x0045EA23);
+        // edi
+        Gfx::RenderTarget& rt = **_renderTarget;
+        auto& drawingCtx = Gfx::getDrawingEngine().getDrawingContext();
+
+        for (auto* ps = (*_paintHead)->basic.nextQuadrantPS; ps != nullptr; ps = ps->nextQuadrantPS)
```
Comment (ZehMatt) [inline src/OpenLoco/src/Paint/Paint.cpp:915]:
```suggestion
        for (const auto* ps = (*_paintHead)->basic.nextQuadrantPS; ps != nullptr; ps = ps->nextQuadrantPS)
```

Comment (ZehMatt) [reply src/OpenLoco/src/Paint/Paint.cpp:915]: Same can be done to the rest as well, we are not modifying those.

**Comment by AaronVanGeffen** [comment]
Produced the hackiest benchmark based on the FPS counter:
```diff
diff --git a/src/OpenLoco/src/Drawing/FPSCounter.cpp b/src/OpenLoco/src/Drawing/FPSCounter.cpp
index 993dc7e6..a441f7ae 100644
--- a/src/OpenLoco/src/Drawing/FPSCounter.cpp
+++ b/src/OpenLoco/src/Drawing/FPSCounter.cpp
@@ -14,12 +14,16 @@ namespace OpenLoco::Drawing
     using TimePoint_t = Clock_t::time_point;
 
     static TimePoint_t _referenceTime;
+    static TimePoint_t _referenceTime10s;
     static uint32_t _currentFrameCount;
+    static uint32_t _currentFrameCount10s;
     static float _currentFPS;
+    static float _currentFPS10s;
 
     static float measureFPS()
     {
         _currentFrameCount++;
+        _currentFrameCount10s++;
 
         auto currentTime = Clock_t::now();
         auto elapsed = std::chrono::duration_cast<std::chrono::milliseconds>(currentTime - _referenceTime).count() / 1000.0;
@@ -31,6 +35,24 @@ namespace OpenLoco::Drawing
             _referenceTime = currentTime;
         }
 
+        static bool init10s = false;
+        if (!init10s)
+        {
+            printf("Initialising benchmark reference point\n");
+            _referenceTime10s = currentTime;
+            init10s = true;
+            return _currentFPS;
+        }
+
+        auto elapsed10s = std::chrono::duration_cast<std::chrono::milliseconds>(currentTime - _referenceTime10s).count() / 1000.0;
+        if (elapsed10s > 10.0)
+        {
+            _currentFPS10s = _currentFrameCount10s / elapsed10s;
+            printf("Produced %d frames in %0.1f (average %f)\n", _currentFrameCount10s, elapsed10s, _currentFPS10s);
+            _currentFrameCount10s = 0;
+            _referenceTime10s = currentTime;
+        }
+
         return _currentFPS;
     }
```

On master, this gets me:
```
$ ninja && ./OpenLoco config/save/Randomland\ by\ matsku84/Randomland\ by\ matsku84\ -\ 2079.SV5 | grep Produced
Produced 596 frames in 10.0 (average 59.594040)
Produced 652 frames in 10.0 (average 65.193481)
Produced 540 frames in 10.0 (average 53.978409)
Produced 605 frames in 10.0 (average 60.415417)
```

**Pull request merged**